### PR TITLE
new CreateOrder msg

### PR DESCRIPTION
**Pull request by politeWall**
- Introducing the new `price_pair` field in the order struct.
- Implementing a new `create_order` message.
- Adding new unit-tests for the updated functionality.
- Updating the contract documentation.
- Updating the front-end script.

### CODE REVIEW

File: src/msg/execute_msg.rs [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
+use crate::types::{OrderPrice, OrderType, SwapAmountInRoute};
 use cosmwasm_schema::cw_serde;
-use cosmwasm_std::Coin;
 
 #[cw_serde]
 pub enum ExecuteMsg {
     CreateOrder {
         order_type: OrderType,
-        order_price: Coin,
+        order_source_denom: String,
```
Comment (cosmic-vagabond) [inline src/msg/execute_msg.rs:8]: it looks like you are using both `source` and `base` to talk about the same thing, can you have one same word you use across all your PR

**Review by cosmic-vagabond — CHANGES_REQUESTED** [review]

File: src/msg/execute_msg.rs (hunk 1 above)
Comment (politeWall) [reply src/msg/execute_msg.rs:8]: if you're talking about the confusion with the error message i just fixed it

Comment (cosmic-vagabond) [reply src/msg/execute_msg.rs:8]: @politeWall nope I am talking about `order_source_denom` vs order base denom it looks like you are using both source and base to refer to the same thing

Comment (politeWall) [reply src/msg/execute_msg.rs:8]: no `order_source_denom` is the denom of the asset that the user send to the contract but `base_denom` from order price is representing what we will be willing buying and selling for the quote asset

Comment (cosmic-vagabond) [reply src/msg/execute_msg.rs:8]: @politeWall understood it makes more sense now once is for indicating the denom of the token used for processing the order, the other is for the pair base/quote denoms used for defining the order price

**Review by cosmic-vagabond — APPROVED** [review]

**Pull request merged**